### PR TITLE
add conditional check

### PR DESCRIPTION
**Pull request by andrewkmin**
## Summary & Motivation
https://github.com/tkhq/sdk/issues/123

## How I Tested These Changes
Examples

### CODE REVIEW

File: packages/http/src/webauthn.ts [hunk 1]
```diff
@@ -131,7 +136,27 @@ export async function getWebAuthnAssertion(
...
+// For additional details see https://web.dev/articles/passkey-form-autofill#feature-detection, https://github.com/w3c/webauthn/wiki/Explainer:-WebAuthn-Conditional-UI
+async function hasWebAuthnSupport(): Promise<boolean> {
+  if (
+    window.PublicKeyCredential &&
+    PublicKeyCredential.isConditionalMediationAvailable
+  ) {
+    const isCMA = await PublicKeyCredential.isConditionalMediationAvailable();
+
+    return isCMA;
+  }
```
Comment (r-n-o) [inline packages/http/src/webauthn.ts:159]: I think this is too restrictive? `isConditionalMediationAvailable` is specific to the autofill feature, which isn't needed for barebone webauthn/passkey support. In our case it's probably enough to return `window.PublicKeyCredential !== undefined`?

File: packages/http/src/webauthn.ts [hunk 2]
```diff
@@ -131,7 +136,20 @@ export async function getWebAuthnAssertion(
...
+
   const res = await webauthnCredentialCreate(options);
 
   return toInternalAttestation(res.toJSON());
 }
+
+// `hasWebAuthnSupport` checks for barebones webauthn support.
+// For additional details and granular settings, see:
+// https://web.dev/articles/passkey-form-autofill#feature-detection, https://developer.mozilla.org/en-US/docs/Web/API/PublicKeyCredential
+function hasWebAuthnSupport(): boolean {
```
Comment (andrewkmin) [inline packages/http/src/webauthn.ts:153]: a little more verbose, but choosing to have this separate function for now in case we want to modify this shared logic

File: packages/http/src/webauthn.ts [hunk 3]
```diff
@@ -114,10 +114,15 @@ export async function getWebAuthnAssertion(
   payload: string,
   options?: TurnkeyCredentialRequestOptions
 ): Promise<string> {
+  const webAuthnSupported = hasWebAuthnSupport();
+
+  if (!webAuthnSupported) {
+    throw new Error("webauthn is not supported by this browser");
+  }
```
Comment (r-n-o) [inline packages/http/src/webauthn.ts:121]: Nit: you can shorten this to:

```ts
  if (!hasWebAuthnSupport()) {
    throw new Error("webauthn is not supported by this browser");
  }
```

**Review by r-n-o — APPROVED** [review]
